### PR TITLE
Add try except for _maybe_evaluate_static call in IndexPropagation

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132128

Fixes the Inductor max-autotune mode failures of the below models:
- GPT2ForSequenceClassification
- PegasusForConditionalGeneration
- XGLMForCausalLM
- hf_GPT2
- tnt_s_patch16_224
```log
  File "/pytorch/torch/_inductor/index_propagation.py", line 329, in statically_true
    evaluated = self.shape_env._maybe_evaluate_static(
  File "/pytorch/torch/fx/experimental/symbolic_shapes.py", line 1499, in wrapper
    return fn_cache(self, *args, **kwargs)
  File "/pytorch/torch/fx/experimental/symbolic_shapes.py", line 4539, in _maybe_evaluate_static
    vr = var_ranges[k]
torch._dynamo.exc.BackendCompilerFailed: backend='compile_fx_wrapper' raised:
KeyError: m_start
```

The `_maybe_evaluate_static` call in `IndexPropagation` may fail. This PR adds try except following the way in `torch/_inductor/sizevars.py` by adding a common utility function.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/sizevars.py [hunk 1]
```diff
@@ -36,6 +36,29 @@
...
+    try:
+        simplified = shape_env._maybe_evaluate_static(
+            expr,
+            axioms=axioms,
+            var_to_range=var_to_range,
+        )
+        if simplified is not None:
+            return bool(simplified)
+    except Exception:
+        log.debug("Could not simplified %s", expr)
```
Comment (jgong5) [inline torch/_inductor/sizevars.py:57]:
```suggestion
        log.debug("Could not simplify %s", expr)
```

**Review by jgong5 — APPROVED** [review]

**Comment by chunyuan-w** [comment]
Hi @jansel could you please help review this PR?

**Review by jansel — APPROVED** [review]

File: torch/_inductor/sizevars.py [hunk 2]
```diff
@@ -36,6 +36,29 @@
...
+    try:
+        simplified = shape_env._maybe_evaluate_static(
+            expr,
+            axioms=axioms,
+            var_to_range=var_to_range,
+        )
+        if simplified is not None:
+            return bool(simplified)
+    except Exception:
+        log.debug("Could not simplify  %s", expr)
```
Comment (Skylion007) [inline torch/_inductor/sizevars.py:57]:
```suggestion
        log.debug("Could not simplify  %s", expr, exc_info=True)
```
Include exception info too

Comment (chunyuan-w) [reply torch/_inductor/sizevars.py]: Thanks for the suggestion. Updated the code as suggested.

**Pull request closed without merging**